src/main/java/mgtow/Mgtow.java

```diff
@@ -0,0 +1,103 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a new Mgtow application instance.
+     *
+     * @param filePath The file path for storing tasks.
+     */
+    public Mgtow(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (InvalidTaskException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public String processCommand(String fullCommand) {
+        try {
+            String[] commandParts = Parser.parseCommand(fullCommand);
+            switch (commandParts[0]) {
+                case "bye":
+                    return ui.getGoodbyeMessage();
+                case "list":
+                    return ui.getTaskListString(tasks.getTasks());
+                case "mark":
+                    tasks.markTask(Integer.parseInt(commandParts[1]) - 1);
+                    storage.saveTasks(tasks.getTasks());
+                    return ui.getMarkTaskMessage(tasks.getTasks().get(Integer.parseInt(commandParts[1]) - 1));
+                case "unmark":
+                    tasks.unmarkTask(Integer.parseInt(commandParts[1]) - 1);
+                    storage.saveTasks(tasks.getTasks());
+                    return ui.getUnmarkTaskMessage(tasks.getTasks().get(Integer.parseInt(commandParts[1]) - 1));
+                case "delete":
+                    Task deletedTask = tasks.deleteTask(Integer.parseInt(commandParts[1]) - 1);
+                    storage.saveTasks(tasks.getTasks());
+                    return ui.getDeleteTaskMessage(deletedTask, tasks.getTasks().size());
+                case "todo":
+                case "deadline":
+                case "event":
+                    Task newTask = Parser.createTask(commandParts);
+                    tasks.addTask(newTask);
+                    storage.saveTasks(tasks.getTasks());
+                    return ui.getAddTaskMessage(newTask, tasks.getTasks().size());
+                case "find":
+                    ArrayList<Task> foundTasks = tasks.findTasks(commandParts[1]);
+                    return ui.getFoundTasksMessage(foundTasks, commandParts[1]);
+                default:
+                    throw new InvalidTaskException("What you talking?");
+            }
+        } catch (InvalidTaskException e) {
+            return ui.getErrorMessage(e.getMessage());
+        } catch (NumberFormatException e) {
+            return ui.getErrorMessage("Invalid number format. Please enter a valid number.");
+        }
+    }
+
+    public void run() {
+        ui.showWelcome();
+        boolean isFinished = false;
+        while (!isFinished) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                String response = processCommand(fullCommand);
+                ui.showMessage(response);
+                if (fullCommand.equals("bye")) {
```

Comment:
Should this take into consideration the capitalisation of the "bye" message? For example, if the user inputs "Bye", this would not evaluate to true